src/OrchardCore.Modules/OrchardCore.Contents/Workflows/Activities/ContentForEachTask.cs

```diff
@@ -0,0 +1,95 @@
+using Microsoft.Extensions.Localization;
+using OrchardCore.Workflows.Abstractions.Models;
+using OrchardCore.Workflows.Activities;
+using OrchardCore.Workflows.Models;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Records;
+using YesSql;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using OrchardCore.ContentManagement.Workflows;
+
+namespace OrchardCore.Contents.Workflows.Activities
+{
+    public class ContentForEachTask : TaskActivity
+    {
+        readonly IStringLocalizer S;
+        private readonly ISession _session;
+        private readonly IContentManager _contentManager;
+
+        public ContentForEachTask(IContentManager contentManager, IStringLocalizer<RetrieveContentTask> localizer, ISession session)
+        {
+            S = localizer;
+            _session = session;
+            _contentManager = contentManager;
+        }
+        public override string Name => nameof(ContentForEachTask);
+
+        public override LocalizedString DisplayText => S["Content For Each Task"];
+
+        public override LocalizedString Category => S["Content"];
+
+        public override IEnumerable<Outcome> GetPossibleOutcomes(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            return Outcomes(S["Iterate"], S["Done"]);
+        }
+        public override bool CanExecute(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            return true;
+        }
+        public override async Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            var contentItems = await _session
+                .Query<ContentItem, ContentItemIndex>(index => index.ContentType == ContentType && index.Published == true && index.Latest == true)
+                .ListAsync();
+            List<ContentItem> items = contentItems.ToList();
+            if (contentItems.Count() == 0)
+            {
+                throw new InvalidOperationException($"The '{nameof(ContentForEachTask)}' failed to evaluate the 'ContentType'.");
+            }
+
+            if (Index < contentItems.Count())
+            {
+                var contentItem = await _contentManager.LoadAsync(items[Index]);
```

Comment:
So you are loading all content items of a type, in a loop, then a single item?

Reply:
I would suggest to at least use `LIMIT 1 WHERE ID > LastItemId` which would be store in the state. OR even better, load `LIMIT 2 WHERE ID >= NextItemId` then you store the second one as NextItemId. Each loop then only load 2 documents max. You could even just return the ids, and load the first one in a separate query to optimize the bandwidth.

Reply:
> So you are loading all content items of a type, in a loop, then a single item?
It was my understanding that LoadAsync triggers the load events and does not make a db call.
![image](https://user-images.githubusercontent.com/89110166/172959065-bf2b47aa-68f8-4759-ae5e-88a5ed4934a0.png)

Reply:
Take parameter added